packages/sdk/src/cosmwasmclient.searchtx.spec.ts

```diff
@@ -0,0 +1,361 @@
+      const client = new CosmWasmClient(wasmdEndpoint);
+      const query = { id: postedSend.hash };
+
+      {
+        const result = await client.searchTx(query, { minHeight: 0 });
+        expect(result.length).toEqual(1);
+      }
+
+      {
+        const result = await client.searchTx(query, { minHeight: postedSend.height - 1 });
+        expect(result.length).toEqual(1);
+      }
+
+      {
+        const result = await client.searchTx(query, { minHeight: postedSend.height });
+        expect(result.length).toEqual(1);
+      }
+
+      {
+        const result = await client.searchTx(query, { minHeight: postedSend.height + 1 });
+        expect(result.length).toEqual(0);
+      }
+    });
+  });
+
+  describe("with SearchByHeightQuery", () => {
+    it("can search by height", async () => {
+      pendingWithoutWasmd();
+      assert(postedSend, "value must be set in beforeAll()");
+      const client = new CosmWasmClient(wasmdEndpoint);
+      const result = await client.searchTx({ height: postedSend.height });
+      expect(result.length).toEqual(1);
+      expect(result[0]).toEqual(
+        jasmine.objectContaining({
+          height: postedSend.height.toString(),
+          txhash: postedSend.hash,
+          tx: postedSend.tx,
+        }),
+      );
+    });
+  });
+
+  describe("with SearchBySentFromOrToQuery", () => {
+    it("can search by sender", async () => {
+      pendingWithoutWasmd();
+      assert(postedSend, "value must be set in beforeAll()");
+      const client = new CosmWasmClient(wasmdEndpoint);
+      const results = await client.searchTx({ sentFromOrTo: postedSend.sender });
+      expect(results.length).toBeGreaterThanOrEqual(1);
+
+      // Check basic structure of all results
+      for (const result of results) {
+        const msg = fromOneElementArray(result.tx.value.msg);
+        assert(isMsgSend(msg), `${result.txhash} (height ${result.height}) is not a bank send transaction`);
+        expect(
+          msg.value.to_address === postedSend.sender || msg.value.from_address == postedSend.sender,
+        ).toEqual(true);
+      }
+
+      // Check details of most recent result
+      expect(results[results.length - 1]).toEqual(
+        jasmine.objectContaining({
+          height: postedSend.height.toString(),
+          txhash: postedSend.hash,
+          tx: postedSend.tx,
+        }),
+      );
+    });
+
+    it("can search by recipient", async () => {
+      pendingWithoutWasmd();
+      assert(postedSend, "value must be set in beforeAll()");
+      const client = new CosmWasmClient(wasmdEndpoint);
+      const results = await client.searchTx({ sentFromOrTo: postedSend.recipient });
+      expect(results.length).toBeGreaterThanOrEqual(1);
+
+      // Check basic structure of all results
+      for (const result of results) {
+        const msg = fromOneElementArray(result.tx.value.msg);
+        assert(isMsgSend(msg), `${result.txhash} (height ${result.height}) is not a bank send transaction`);
```

Comment:
Yeah, this is the question I posed below - do we want to only match MsgSend, or other transactions that also affect the balance?

I am fine with just matching MsgSend if this is what we want, but then adjust the queries to be only that